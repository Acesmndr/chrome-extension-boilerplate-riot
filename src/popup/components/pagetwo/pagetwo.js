import helper from '../../helper';
import { sendMessage } from '../../../chrome/chrome-utils';

export default {
  state: {
    todolist: [],
    containsItem: false,
  },
  addItem() {
    const item = this.$('input[type="text"]').value;
    if (item === '') {
      return;
    }
    sendMessage({
      type: 'saveDataInBackground',
      data: { todolist: [...this.state.todolist, item] },
    });
    this.$('input[type="text"]').value = '';
  },
  removeItem(e) {
    let itemList = this.state.todolist;
    debugger;
    itemList.splice(Number(e.target.getAttribute('itemNo')), 1);
    sendMessage({
      type: 'saveDataInBackground',
      data: { todolist: itemList },
    });
  },
  clearAll() {
    sendMessage({
      type: 'removeAllDataInBackground',
    })
  },
  goToFirstPage() {
    helper.route('pageone');
  },
  onBeforeMount(props, state) {
    sendMessage({
      type: 'getDataFromBackground',
      query: 'todolist'
    });
  },
  onBeforeUpdate(props, state) {
    console.log(props, state);
    state = {
      todolist: props.todolist || [],
      containsItem: props.containsItem || false,
    }
  }
}