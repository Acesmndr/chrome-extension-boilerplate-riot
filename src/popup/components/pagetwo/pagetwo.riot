<pagetwo>
	<div class='card'>
		<div class='content'>
			<div class='form'>
				<div class='title'>To Do</div>
				<ul class='todolist'>
					<li each = { (todoItem, i) in (state.todolist || []) } onclick={ removeItem } itemNo={i}>{ todoItem }</li>
				</ul>
				<img src='assets/img/empty.svg' class='empty' if={ !state.todolist || !state.todolist.length } />
				<input type='text' required placeholder='Add item' autofocus/>
				<input type='button' class='add-btn' value='+' onclick={ addItem } />
			</div>
		</div>
		<div class="content right-align buttons">
				<input type='button' class='clear' onclick={ clearAll } value='Clear All'/>
			<input type='button' onclick={ goToFirstPage } value='Return to page 1'/>
		</div>
	</div>
	<style type='scss'>
		@import './src/popup/components/pagetwo/pagetwo';
	</style>
	<script>
		import script from './pagetwo.js';
		export default script;
	</script>
</pagetwo>
