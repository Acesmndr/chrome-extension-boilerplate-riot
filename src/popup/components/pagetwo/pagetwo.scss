@import '../../style/colorpalette';

.form {
  font-size: 12px;
  .title {
    margin-bottom: 24px;
    color: $simply-grey;
    text-align: center;
    font-size: 22px;
  }
  .empty {
    margin-bottom: 32px;
    width: 320px;
  }
  input[type='text'] {
    padding: 10px;
    width: 260px;
    border: 1px solid $simply-grey;
    border-radius: 2px 0 0 2px;
    font-size: 16px;
  }
  .add-btn {
    background: $simply-green;
    border-radius: 0 2px 2px 0;
    margin: 0;
    font-size: 24px;
    line-height: 24px;
    padding: 5px 12px 11px 10px;
  }
}

.todolist {
  li {
    font-size: 20px;
    margin-bottom: 24px;
    color: $simply-grey;
    &:hover {
      text-decoration: line-through;
    }
  }  
}

.right-align.buttons {
  margin-top: 32px;
}
