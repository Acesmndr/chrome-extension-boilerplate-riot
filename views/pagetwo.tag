<pagetwo>
	<div class='card'>
		<div class='content'>
			<form>
				<label for='name-input'>Enter a todo</label>
				<input type='text' id='name-input'/>
				<input type='button' value='Add' />
			</form>
		</div>
		<div class="content right-align">
			<input type='button' onclick={ pageRedirect } value='Return to page 1'/>
		</div>
	</div>
	<style type='scss'>
		@import 'styles/pagetwo';
	</style>
	<script src="presenters/pagetwo.js"></script>
</pagetwo>