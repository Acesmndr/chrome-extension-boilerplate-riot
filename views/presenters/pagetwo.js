this.goToFirstPage = () => {
  riot.mount('.main-body', 'pageone');
}

this.on('mount', () => {
  this.mixin('messageMixin');
  this.setupListener(this);
  chrome.runtime.sendMessage({
    type: 'getDataFromBackground',
  });
});

this.on('unmount', () => {
  console.log('Blank got unmounted');
});