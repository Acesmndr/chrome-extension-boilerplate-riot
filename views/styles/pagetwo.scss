@import './colorpalette';

.form {
  padding: 16px;
  font-size: 16px;
  text-align: center;
  .title {
    margin-bottom: 32px;
    color: $simply-blue;
    font-weight: bold;
    font-size: 32px;
  }
  input[type='text'] {
    padding: 10px;
    font-size: 16px;
  }
}

.todolist {
  li {
    font-size: 24px;
    color: #7a818d;
    &:hover {
      text-decoration: line-through;
    }
  }  
}