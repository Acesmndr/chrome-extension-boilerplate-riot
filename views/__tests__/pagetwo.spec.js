import * as riot from 'riot';
import pagetwo from '../pagetwo.tag';

const mockFn = jest.fn();
const messageMixin = {
    setupListener: (e) => new mockFn(e),
    sendMessage: (msg, callback = () => {}) => new mockFn(msg),
    notify: (params, callback) => new mockFn(params, callback),
}; 
describe('Page two specs', () => {
    beforeAll( () => {
        expect(mockFn.mock.calls.length).toBe(0);
        riot.mixin('messageMixin', messageMixin);
        // create mounting point 
        const elem = document.createElement('div');
        elem.className = "main-body";
        document.body.appendChild(elem)
        const opts = { data: ['batman', 'superman'] };
        riot.mount(elem, 'pagetwo', opts);
    });
    
    
    it('should mount the tag', () => {
        expect(document.querySelector('.main-body')).toMatchSnapshot();
    });
    
    it('should setup a listener on mount', () => {
        expect(mockFn.mock.calls.length).toBe(2);
    });

    it('should have a title text', () => {
        expect(document.querySelector('.form .title').textContent).toBe('To Do App');
    });

    it('should have a input text field', () => {
        expect(document.querySelectorAll('input[type="text"]').length).toBe(1);
    });

    it('should have two fields set with opts recieved', () => {
        expect(document.querySelectorAll('ul li').length).toBe(2);
    });

    it('should remove an item from todo list on click', () => {
      const firstUl = document.querySelectorAll('ul li')[0];
      firstUl.click();
      expect(document.querySelectorAll('ul li').length).toBe(1);
      expect(mockFn.mock.calls.length).toBe(3);
      expect(mockFn.mock.calls[2]).toEqual([{ type: 'saveDataInBackground', data: {todolist: ['superman']} }]);
    });

    it('should add an item on todo list', () => {
      expect(document.querySelectorAll('ul li').length).toBe(1);
      document.querySelector('input[type="text"]').value="wonderwoman";
      const addButton = document.querySelectorAll('input[type="button"]')[0];
      addButton.click();
      expect(document.querySelectorAll('ul li').length).toBe(2);
      expect(mockFn.mock.calls.length).toBe(4);
      expect(mockFn.mock.calls[3]).toEqual([{ type: 'saveDataInBackground', data: {todolist: ['superman', 'wonderwoman']} }]);
    });
    
    it('should send a clear request on button click', () => {
        const clearButton = document.querySelectorAll('input[type="button"]')[1];
        clearButton.click();
        expect(mockFn.mock.calls.length).toBe(5);
        expect(mockFn.mock.calls[4]).toEqual([{ type: 'removeAllDataInBackground' }]);
    });

    it('should call a mount function to mount the first page on click', () => {
        riot.mount = (where, what) => new mockFn(where, what);
        const routeButton = document.querySelectorAll('input[type="button"]')[2];
        routeButton.click();
        expect(mockFn.mock.calls.length).toBe(6);
        expect(mockFn.mock.calls[5]).toEqual(['.main-body', 'pageone']);
    });


});