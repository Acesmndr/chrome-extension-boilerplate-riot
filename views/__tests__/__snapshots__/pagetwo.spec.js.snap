// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page two specs should mount the tag 1`] = `
<div
  class="main-body"
  data-is="pagetwo"
>
  <div
    class="card"
  >
     
    <div
      class="content"
    >
       
      <div
        class="form"
      >
         
        <div
          class="title"
        >
          To Do App
        </div>
         
        <input
          autofocus=""
          ref="nameInput"
          required=""
          type="text"
        />
         
        <input
          type="button"
          value="Add"
        />
         
      </div>
       
      <ul
        class="todolist"
      >
         
        <li>
          batman
        </li>
        <li>
          superman
        </li>
        
         
      </ul>
       
    </div>
     
    <div
      class="content right-align"
    >
       
      <input
        type="button"
        value="Clear All"
      />
       
      <input
        type="button"
        value="Return to page 1"
      />
       
    </div>
     
  </div>
</div>
`;
